From 386394fc9fcde2615391f804eaa466749f96f4ef Mon Sep 17 00:00:00 2001
From: Lasse Collin <lasse.collin@tukaani.org>
Date: Sat, 1 Jun 2019 21:36:13 +0300
Subject: [PATCH] liblzma: memcmplen: Use ctz32() from tuklib_integer.h.

The same compiler-specific #ifdefs are already in tuklib_integer.h
---
 src/liblzma/common/memcmplen.h | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/src/liblzma/common/memcmplen.h b/src/liblzma/common/memcmplen.h
index 62e7983..25debf6 100644
--- a/src/liblzma/common/memcmplen.h
+++ b/src/liblzma/common/memcmplen.h
@@ -99,15 +99,7 @@ lzma_memcmplen(const uint8_t *buf1, const uint8_t *buf2,
 			_mm_loadu_si128((const __m128i *)(buf2 + len))));
 
 		if (x != 0) {
-#	if defined(__INTEL_COMPILER)
-			len += _bit_scan_forward(x);
-#	elif defined(_MSC_VER)
-			unsigned long tmp;
-			_BitScanForward(&tmp, x);
-			len += tmp;
-#	else
-			len += __builtin_ctz(x);
-#	endif
+			len += ctz32(x);
 			return my_min(len, limit);
 		}
 
